openedx/openedx-platform
# [BD-04] Convert ErrorModule and NonStaffErrorModule to XBlocks. [SE-3191]

Converts the Error XModule and NonStaffError XModule into XBlocks.

Part of [XModule to XBlock Conversion work](https://openedx.atlassian.net/wiki/spaces/AC/pages/1472790755/XModule+to+XBlock+Conversion). 

**Testing instructions**:

* In a course add a video xblock and an annotatable xmodule ([howto](https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/annotation.html)) and publish them.
* Cherry-pick the second commit in this PR with the message: "Temp code for testing.".
* View the xblock and xmodule in studio and LMS and note the behavior.
* Check out this branch and check the behavior of both in studio and LMS again.

**MRO Analysis**

Descriptor MRO | Module MRO | Block MRO | Notes
------------ | ------------- | ---------------- | ------------
<class 'xmodule.error_module.ErrorDescriptor'> | <class 'xmodule.error_module.ErrorModule'> | <class 'xmodule.error_module.ErrorBlock'>
<class 'xmodule.error_module.ErrorFields'> | <class 'xmodule.error_module.ErrorFields'> | <class 'xmodule.error_module.ErrorFields'>
<class 'xmodule.x_module.XModuleDescriptor'> | |
<class 'xmodule.x_module.XModuleDescriptorToXBlockMixin'> | | <class 'xmodule.x_module.XModuleDescriptorToXBlockMixin'>
 | <class 'xmodule.x_module.XModule'> | 
 | <class 'xmodule.x_module.XModuleToXBlockMixin'> | <class 'xmodule.x_module.XModuleToXBlockMixin'> | <class 'xmodule.x_module.XModuleToXBlockMixin'>
<class 'xmodule.x_module.HTMLSnippet'> | <class 'xmodule.x_module.HTMLSnippet'> | <class 'xmodule.x_module.HTMLSnippet'>
<class 'xmodule.x_module.ResourceTemplates'> | | <class 'xmodule.x_module.ResourceTemplates'>
<class 'xmodule.x_module.XModuleMixin'> | <class 'xmodule.x_module.XModuleMixin'> | <class 'xmodule.x_module.XModuleMixin'>
<class 'xmodule.x_module.XModuleFields'> | <class 'xmodule.x_module.XModuleFields'> | <class 'xmodule.x_module.XModuleFields'>
<class 'xblock.core.XBlock'> | <class 'xblock.core.XBlock'> | <class 'xblock.core.XBlock'>


### Annotatable XModule

In Studio:

![image](https://user-images.githubusercontent.com/5305552/100030281-f20b9100-2e14-11eb-965b-2b4ff8e6eedd.png)

In LMS with staff account (component will not be displayed for non-staff accounts):

![image](https://user-images.githubusercontent.com/5305552/100030208-cab4c400-2e14-11eb-964b-f35c8af13474.png)

### Video XBlock

In Studio:

![image](https://user-images.githubusercontent.com/5305552/100030624-c8069e80-2e15-11eb-94ec-fe571c85428e.png)

In LMS with staff account (component will not be displayed for non-staff accounts):

![image](https://user-images.githubusercontent.com/5305552/100030567-a1486800-2e15-11eb-98be-9cb0fff0a8f9.png)

This does seem to be unexpected behavior but I have left it unchanged.
